### PR TITLE
Breakout npm_and_yarn, Move some dockerfiles

### PR DESCRIPTION
**Pull request by pavera**
This breaks out npm_and_yarn and also moves the previously broken out ecosystem's dockerfiles into their respective subdirs to avoid clutter at the top level.

### CODE REVIEW

File: out.yml [hunk 1]
```diff
@@ -0,0 +1,897 @@
+input:
```
Comment (jakecoffman) [inline out.yml:1]: Did you mean to commit this file?

Comment (pavera) [reply out.yml:1]: no...

**Comment by pavera** [comment]
yeah, I'm digging to try to figure it out, I'll get it soon I'm sure.

**Review by jakecoffman — APPROVED** [review]
Nice work!

**Pull request merged**